### PR TITLE
add explainer

### PR DESCRIPTION
**Pull request by philloooo**

### CODE REVIEW

File: explainers.md [hunk 1]
```diff
@@ -0,0 +1,860 @@
...
+
+#### Android implementations
+
+When discussing behaviors on Android, this doc is exclusively only covering Chromium-based browsers on Android.
+
+
+#### Desktop implementations
+
+When discussing behaviors on desktop, this doc covers Chromium-based browsers and Firefox (desktop).
+
```
Comment (dmurph) [inline explainers.md:234]: It might be worth adding iOS implementation here too. Asking around now to see what that is.

File: explainers.md [hunk 2]
```diff
@@ -0,0 +1,860 @@
...
+*   [S2: installed apps on desktop](#scenario-2), [S3: Apps that are installed on both Android and desktop](#scenario-3) needs migration.
+*   Change of implicit expectations for desktop WebApp developers
+*   **[S6: Apps that specify manifest as data URL](#scenario-6) will not be updatable**
+*   PWA stores will recognize different versions of manifest URLs as different web apps. Alternatively they can load the start\_url to find the manifest link, and follow redirect to find the latest manifest\_url and use that as the authoritative manifest. 
+*   [S4: Apps that need to update manifest URLs](#scenario-4) will need to keep supporting the previous manfiest\_url, handle 301 to the latest manifest forever. The document will need to always use the original manifest\_url. An extra roundtrip will always be needed for loading the app.
+*   If the manifest\_url is hosted under a CDN, they will need to stick to that CDN provider.
+*   Difficult to reconcile with [S8: apps installed from A2HS](#scenario-8), which does NOT require a manifest. ID creating here is weird, and instead would have to have custom upgrade handling for when a manifest is found on the page.
+
+
```
Comment (dmurph) [inline explainers.md:737]: Might be nice to add a 

Note: If there weren't existing apps & infrastructure that relied on a changing manifest_url, and this was being designed from the beginning, this option is a lot more attractive.

File: explainers.md [hunk 3]
```diff
@@ -0,0 +1,860 @@
...
+Using **start\_url\_origin + id** would be preferred as this is the most stable option.
+
+For the default field to use, **start\_url** is preferred.
+
+Because of the existing sync design of desktop PWAs, old chromes will be unavoidable to have duplicate apps if the default is not start\_url, there might be weird behaviors when duplicate apps exist.
+
+Due to the use cases for data URL, cross origin manifest URL, as well as the sync behavior, using start\_url is preferred over manfiest\_url as default.
+
+
```
Comment (dmurph) [inline explainers.md:853]: Might be nice to acknowledge the discoverability issue, and welcome ideas to address this.

**Review by dmurph — APPROVED** [review]
LGTM with small nits

File: explainers.md (hunk 3 above)
Comment (philloooo) [reply explainers.md:853]: Mentioned this in the beginning of the conclusion

File: explainers.md (hunk 2 above)
Comment (philloooo) [reply explainers.md:737]: added

**Pull request merged**